edg/parts/Microcontroller_Stm32g431.py

```diff
@@ -0,0 +1,273 @@
+from typing import *
+
+from .JlcPart import JlcPart
+from ..abstract_parts import *
+
+
+@abstract_block
+class Stm32g431Base_Device(IoControllerI2cTarget, IoControllerCan, IoControllerUsb, InternalSubcircuit,
+                           BaseIoControllerPinmapGenerator,
+                           GeneratorBlock, JlcPart, FootprintBlock):
+    PACKAGE: str  # package name for footprint(...)
+    PART: str  # part name for footprint(...)
+    LCSC_PART: str
+    LCSC_BASIC_PART: bool
+
+    SYSTEM_PIN_REMAP: Dict[str, Union[str, List[str]]]  # pin name in base -> pin name(s)
+    RESOURCE_PIN_REMAP: Dict[str, str]  # resource name in base -> pin name
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+        # Power and ground
+        self.pwr = self.Port(VoltageSink(
+            voltage_limits=(1.71, 3.6) * Volt,
+            current_draw=(0.001, 150.0) * mAmp  # table 15
```

Comment:
150mA is a max limit, not the actual draw of the device. Table 21 gives the core running current, which specifies a worst-case max of 44mA (at a toasty 125C) and low of 14nA in sleep mode (Table 32).

At some point it would probably be worth modeling these limits (total current into device pins, total IO current), but we don't currently.